go/vt/discovery/tablet_health_check.go

```diff
@@ -272,8 +271,16 @@ func (thc *tabletHealthCheck) checkConn(hc *HealthCheckImpl) {
 			}
 		}()
 
+		// Limit the number of healthcheck connections opened in parallel to avoid high OS-thread
+		// usage due to blocking networking syscalls (eg: DNS lookups, TCP connection opens,
+		// etc). Without this limit it is possible for vtgates watching >10k tablets to hit
+		// the panic: 'runtime: program exceeds 10000-thread limit'.
+		hc.healthCheckConnectSem <- 1
+		conn := thc.Connection()
+		<-hc.healthCheckConnectSem
```

Comment:
We can try this, but I am not sure this will help because my understanding is grpc sets up connections lazily on first RPC call. Unless you force it with the [WithBlock dial option](https://pkg.go.dev/google.golang.org/grpc#WithBlock) which is discouraged (and not used in this context based on a code read)